crates/project/src/project.rs

```diff
@@ -5151,10 +5157,20 @@ impl Project {
         };
 
         if let Some(lsp_documentation) = completion_item.documentation {
+            let base_width =
+                (10.max((-2.8333 * buffer_font_size as f32 + 87.0) as i32) as f32) as i32;
+
+            let remaining_width = base_width - (completion_item.label.len() as f32 - 30.0) as i32;
+
+            let max_documentation_length: usize = 10.max(
+                (remaining_width as f32 * buffer_font_size as f32 / (ui_font_size as f32)) as usize,
+            );
+
```

Comment:
Again, I don't think this part of the code should be concerned with presentation details like truncation. We should do this in the UI layer, when rendering completions.